src/machine/machined-varlink.c

```diff
@@ -415,32 +417,78 @@ static int list_machine_one(sd_varlink *link, Machine *m, bool more) {
         return sd_varlink_reply(link, v);
 }
 
+typedef struct MachineListQueryParams {
+    const char *machine_name;
+    pid_t pid;
```

Comment:
Please use 8ch indentation.

Not a matter, but I think it is OK to move the declaration of the type into the function. Then, maybe you can shorten the type name.